osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs

```diff
@@ -51,5 +62,36 @@ protected override void LoadComplete()
                 circle.ResizeTo(new Vector2(0.8f), 500, EasingTypes.OutQuint);
             }
         }
+
+        private class CapsWarning : TextAwesome, IHasTooltip
+        {
+            public string TooltipText => Console.CapsLock ? @"Caps lock is active" : string.Empty;
+
+            public override bool HandleInput => true;
+
+            public CapsWarning()
+            {
+                Icon = FontAwesome.fa_warning;
+                Origin = Anchor.CentreRight;
+                Anchor = Anchor.CentreRight;
+                Margin = new MarginPadding { Right = 10 };
+                AlwaysPresent = true;
+                Alpha = 0;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colour)
+            {
+                Colour = colour.YellowLight;
+            }
+
+            protected override void Update()
+            {
+                base.Update();
+                updateVisibility();
+            }
+
+            private void updateVisibility() => FadeTo(Console.CapsLock ? 1 : 0, 250, EasingTypes.OutQuint);

```

Comment:
you can't do this with transform helpers. you need to store a state and run them only once, else they are going to run far longer than you expect.